x/concentrated-liquidity/swaps.go

```diff
@@ -362,13 +362,27 @@ func (k Keeper) computeOutAmtGivenIn(
 			if err != nil {
 				return sdk.Coin{}, sdk.Coin{}, 0, sdk.Dec{}, sdk.Dec{}, sdk.Dec{}, err
 			}
+			// Update the swapState's tick with the tick we retrieved liquidity from
+			swapState.tick = swapStrategy.UpdateTickAfterCrossing(nextTick)
 		} else if !sqrtPriceStart.Equal(computedSqrtPrice) {
 			// Otherwise if the sqrtPrice calculated from ComputeSwapWithinBucketOutGivenIn(...) does not equal the sqrtPriceStart we started with at the
 			// beginning of this iteration, we set the swapState tick to the corresponding tick of the computedSqrtPrice calculated from ComputeSwapWithinBucketOutGivenIn(...)
 			price := computedSqrtPrice.Mul(computedSqrtPrice)
-			swapState.tick, err = math.PriceToTickRoundDown(price, p.GetTickSpacing())
-			if err != nil {
-				return sdk.Coin{}, sdk.Coin{}, 0, sdk.Dec{}, sdk.Dec{}, sdk.Dec{}, err
+			newTick := math.CalculatePriceToTick(price)
+
+			// TEMPORARY HACK: this is to fix tick rounding error where
+			// the tick is off by 1 due to banker's rounding error in CalculatePriceToTick
+			// TODO: if this is to remain in the codebase, consider abstracting this into a
+			// method of swap strategy.
+			isZeroForOne := getZeroForOne(tokenInMin.Denom, p.GetToken0())
+			if isZeroForOne {
+				if newTick <= swapState.tick {
+					swapState.tick = newTick
+				}
+			} else {
+				if newTick > swapState.tick {
+					swapState.tick = newTick
```

Comment:
Are there any cases where none of these branches would be hit?

Reply:
```
			if isZeroForOne {
				if newTick <= swapState.tick {
					swapState.tick = newTick
				} else {
					fmt.Println("11111")
				}
			} else {
				if newTick > swapState.tick {
					swapState.tick = newTick
				} else {
					fmt.Println("22222")
				}
			}
```

Prior to merging #5541 all branches were hit. Now, only "22222" is being hit.

I'm going to make an issue to investigate this separately. Merging for now